src/js/elements/mapchip/legend.js

```diff
@@ -0,0 +1,51 @@
+import {format as d3format} from 'd3-format';
+
+import { Observable } from "../../utils";
+
+const lightStart = 3;
+
+export class Legend extends Observable {
+    constructor(container, legendColors) {
+        super();
+        this.prepareDomElements(container);
+        this.legendColors = legendColors;
+    }
+
+    prepareDomElements(container) {
+        const legendBlock = $('.map_legend-block')[0];
+        this.legendContainer = container;
+        this.clonedLegendBlock = legendBlock.cloneNode(true);	//a legend block
+        this.clonedLegend = $('.map-options__legend')[0].cloneNode(true);	//the legend itself
+
+        this.clear();
+    }
+
+    clear() {
+        
+    }
+
+    show(colors, intervals) {
```

Comment:
I can't remember what intervals is it just a list of numbers?

Reply:
maybe the naming here is not explanatory enough. it's the string array of the number(including the percentage sign) we show in the legend. for example for the attached legend, the intervals is 

`["9.8%", "10.8%", "11.8%", "12.8%", "13.7%"]`

![image](https://user-images.githubusercontent.com/53019884/122735786-0834df00-d288-11eb-8243-036dc8831d32.png)

Reply:
maybe `intervalLabels` then

Reply:
done